modules/core/src/main/scala/org/tessellation/modules/Programs.scala

```diff
@@ -3,30 +3,39 @@ package org.tessellation.modules
 import java.security.KeyPair
 
 import cats.effect.Async
+import cats.effect.std.Random
 
 import org.tessellation.config.types.AppConfig
 import org.tessellation.domain.cluster.programs.TrustPush
 import org.tessellation.infrastructure.snapshot.programs.RollbackLoader
 import org.tessellation.kryo.KryoSerializer
 import org.tessellation.sdk.domain.cluster.programs.{Joining, PeerDiscovery}
+import org.tessellation.sdk.domain.snapshot.PeerSelection
+import org.tessellation.sdk.http.p2p.clients.L0GlobalSnapshotClient
+import org.tessellation.sdk.infrastructure.snapshot.MajorityPeerSelection
 import org.tessellation.sdk.infrastructure.snapshot.programs.Download
 import org.tessellation.sdk.modules.SdkPrograms
 import org.tessellation.security.SecurityProvider
 
+import org.http4s.client.Client
+
 object Programs {
 
-  def make[F[_]: Async: KryoSerializer: SecurityProvider](
+  def make[F[_]: Async: KryoSerializer: SecurityProvider: Random](
     sdkPrograms: SdkPrograms[F],
     storages: Storages[F],
     services: Services[F],
     keyPair: KeyPair,
-    config: AppConfig
+    config: AppConfig,
+    client: Client[F]
   ): Programs[F] = {
     val trustPush = TrustPush.make(storages.trust, services.gossip)
+    val peerSelector: PeerSelection[F] = MajorityPeerSelection.make(storages.cluster, L0GlobalSnapshotClient.make(client))
```

Comment:
I think `L0GlobalSnapshotClient` is already instantiated in `p2pClient`. You can just pass it here instead of creating a new one

Reply:
Also, you have a name mismatch: `peerSelector` vs `PeerSelection`. btw - why not `PeerSelect`?

Reply:
Changed

Reply:
Ok, I was not aware of that. If so, please add `L0GlobalSnapshotClient` to the `P2PClients` itself and pass it here. That way we have a single place where we pass `client: Client[F]` and instantiate all p2p clients.

Reply:
Added.